### PR TITLE
feat(standalone): support optionally including components

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

closes https://github.com/risingwavelabs/risingwave/issues/12383

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]

**Comment by fuyufjh** [comment]
So how to specify which components to start in the command line?

**Comment by kwannoel** [comment]
> So how to specify which components to start in the command line?

If options are specified, for the component e.g. `--meta-node-opts="..."`, we will run the component. If not specified, we don't run it.

**Comment by kwannoel** [comment]
Adding an e2e test case.

File: ci/scripts/standalone-utils.sh [hunk 1]
```diff
@@ -6,9 +6,43 @@ export RW_PREFIX=$PWD/.risingwave
...
+        --role both \
+        --meta-address http://127.0.0.1:5690" \
+     --frontend-opts=" \
+       --listen-addr 127.0.0.1:4566 \
+       --advertise-addr 127.0.0.1:4566 \
+       --prometheus-listener-addr 127.0.0.1:2222 \
+       --health-check-listener-addr 127.0.0.1:6786 \
+       --meta-addr http://127.0.0.1:5690" >"$1" 2>&1
+}
+
```
Comment (kwannoel) [inline ci/scripts/standalone-utils.sh:42]: Example of starting cluster without compactor.

**Pull request merged**